### ISSUE TITLE
Get-RsCatalogItemRole fails when -Identity has no policy; cryptic stack trace 

### ISSUE THREAD

**Issue by jzabroski** [Microsoft/ReportingServicesTools#177]
# Error message
Get-RsCatalogItemRole : Cannot bind argument to parameter 'Policy' because it  is null. 

# Analysis
The problem appears to be here:

https://github.com/Microsoft/ReportingServicesTools/blob/a7836d807f6f4ee169d7d45787cd92be587b1cbf/ReportingServicesTools/Functions/Security/Get-RsCatalogItemRole.ps1#L97-L102

# Suggested Improvement
I suggest the following logic, which will allow the default policy (root folder) when the Credential is an administrator:

```
        # Filter Polices by Identity
        if($Identity) {
            $parentPolicyTmp = $parentPolicy | Where-Object { $_.GroupUserName -eq $Identity }
            if ($parentPolicyTmp -ne $null)
            {
                $parentPolicy = $parentPolicyTmp
            }
        }

        $parentType = $Proxy.GetItemType($Path)
```

**Comment by jtarquino** [comment]
@ClaudioESSilva have you seen this one?

**Comment by NJarZe** [comment]
What is the expected behavior for this fix? Is it to return empty when the user doesn't have access to the path?

**Comment by jzabroski** [comment]
@NJarZe  What do you suggest? Have you run into this also?  Either it should throw an exception with a useful message, or do something similar to what I have suggested.

**Comment by NJarZe** [comment]
@jzabroski I have encountered this issue before, and had to work around it for a very unique use case (granting access by copying another user's exact security settings). 

I think your idea is good, return the parent policy; however, if it's a nested folder, the parent policy will also be empty, no? If so, it might be better to raise an exception with a useful message.

By the way: I'm new to the GitHub community, so I apologize if I'm not following the right procedure... Just trying to contribute in any way to this nice tool I use quite often :)

**Comment by jzabroski** [comment]
I agree, @jtarquino  and his team did a fantastic job building this tool. The clever use of PowerShell to add SOAP types dynamically, to abstract away which version of the API we talk to, is simply ingenius.  I've participated in a lot of bad PowerShell projects, since PowerShell is so easy to write bad code in, but I've gotten a lot of value out of this project.

I think what I will try to do is first reproduce the issue.

Also, I agree, the real reason this is even a problem is that the default way SSRS does permissions is rather bad.  For example, how I lay out my projects is each project is a top-level folder in the final deployment. But then, how do you avoid having different departments see each other's reports?  Well, with the default model, out of the box, you only get the following roles:


| Role | Description |
| ----- | ------------ |
| Browser | May view folders, reports and subscribe to reports. |
| Content Manager | May manage content in the Report Server. This includes folders, reports and resources. |
| My Reports | May publish reports and linked reports; manage folders, reports and resources in a users My Reports folder. |
| Publisher | May publish reports and linked reports to the Report Server. |
| Report Builder | May view report definitions. |

So, the first thing I do in a new SSRS deployment is add a new role:
| Role | Description |
| Folder Browser | May view folders, but cannot view folder contents. Useful if users have Browser role in multiple folders, so that at the top-level they can see all of the folders then can Browse to. |

UNFORTUNATELY, SSRS API does not provide a way to add new Roles.  You have to use SQL to do this:

```sql
USE SqlReport_dev

SET NOCOUNT ON

CREATE TABLE #Roles (
	RoleName NVARCHAR(260) NOT NULL,
	[Description] NVARCHAR(512) NULL,
	TaskMask NVARCHAR(32) NOT NULL
)
	

INSERT INTO #Roles
EXEC dbo.GetRoles @RoleFlags = 0 -- tinyint

IF NOT EXISTS (
	SELECT NULL FROM #Roles WHERE RoleName = 'Folder Browser'
)
BEGIN
	EXEC dbo.CreateRole
		@RoleID='290C8111-820E-4A37-9ABF-CB3B09153C52',
		@RoleName=N'Folder Browser',
		@Description=N'Allow ability to browse folders',
		@TaskMask=N'0000001000000000',
		@RoleFlags=0
END

DROP TABLE #Roles
```

Note: I purposefully hard-code the RoleID GUID so that it is consistent across environments, so that the likelihood of an issue occurring in production due to variances in RoleIDs is unlikely.

**Comment by jzabroski** [comment]
Note: The above SQL fails on SSRS 2017 due to TaskMask bit mask being larger in 2017 than in 2016.  Buyer beware!
